aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -1498,6 +1569,11 @@ module aptos_framework::delegation_pool {
                 pool.observed_lockup_cycle,
                 pool_u64::create_with_scaling_factor(SHARES_SCALING_FACTOR)
             );
+
+            // update the commission percentage for the new lockup cycle
+            if (exists<CommissionPercentage>(pool_address)) {
+                pool.operator_commission_percentage = borrow_global<CommissionPercentage>(pool_address).commission_percentage_next_lockup_cycle;
```

Comment:
There's a slight problem here - if synchronize_delegation_pool is not called, the commission percentage would not be updated. This might not be too big of a deal as user operations such as staking would call synchronize_delegation_pool. However, the UI would still display the old commission rate before any actual user operations. Can you think a bit more on whether there are some options we can use to address this?

Reply:
I updated `operator_commission_percentage` to return the new commission rate if it's available:
```
    public fun operator_commission_percentage(pool_address: address): u64 acquires DelegationPool, NextCommissionPercentage {
        assert_delegation_pool_exists(pool_address);
        if (is_next_commission_percentage_effective(pool_address)) {
            operator_commission_percentage_next_lockup_cycle(pool_address)
        } else {
            borrow_global<DelegationPool>(pool_address).operator_commission_percentage
        }
    }
```
This provides a better UX for those who are adding a new stake to the pool.